product-scenarios/10-manage-security-resiliancy-other-aspects-of-service-endpoints/10.1-expose-service-with-secured-backend/10.1.1-expose-service-secured-via-basic-authentication/src/test/java/org/wso2/am/scenario/tests/basic/auth/security/BasicAuthEndpointSecuredAPITestcase.java

```diff
@@ -0,0 +1,181 @@
+import org.json.JSONObject;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.am.integration.test.utils.APIManagerIntegrationTestException;
+import org.wso2.am.integration.test.utils.base.APIMIntegrationConstants;
+import org.wso2.am.integration.test.utils.bean.APILifeCycleState;
+import org.wso2.am.integration.test.utils.bean.APILifeCycleStateRequest;
+import org.wso2.am.integration.test.utils.bean.APPKeyRequestGenerator;
+import org.wso2.am.integration.test.utils.bean.SubscriptionRequest;
+import org.wso2.am.scenario.test.common.APIPublisherRestClient;
+import org.wso2.am.scenario.test.common.APIRequest;
+import org.wso2.am.scenario.test.common.APIStoreRestClient;
+import org.wso2.am.scenario.test.common.HttpClient;
+import org.wso2.am.scenario.test.common.ScenarioTestBase;
+import org.wso2.carbon.automation.test.utils.http.client.HttpResponse;
+import org.wso2.carbon.apimgt.samples.utils.WebAppDeployUtils;
+
+import javax.ws.rs.core.Response;
+import javax.xml.bind.DatatypeConverter;
+
+import static org.eclipse.jdt.internal.compiler.util.Util.UTF_8;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class BasicAuthEndpointSecuredAPITestcase extends ScenarioTestBase {
+    private static final Log log = LogFactory.getLog(BasicAuthEndpointSecuredAPITestcase.class);
+    private final String apiName = UUID.randomUUID().toString();
+    private final String apiContext = "/" + UUID.randomUUID().toString();
+    private final String admin = "admin";
+    private final String apiVersion = "1.0.0";
+    private final String apiResource = "/sec";
+    private final String applicationName = "EndpointSecurityApplication";
+    private final String applicationDescription = "SampleDescription";
+    private final String endpointType = "secured";
+    private final String endpointAuthType = "basicAuth";
+    private final String epUsername = "wso2user";
+    private final String epPassword = "!@#$%^wso2.123$%";
+    private final String endpointURL = "https://localhost:9443/jaxrs_basic/services/customers/customerservice/";
+
+    private APIRequest apiRequest;
+    private Properties infraProperties;
+    private APIPublisherRestClient apiPublisher;
+    private APIStoreRestClient apiStore;
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws APIManagerIntegrationTestException {
+        infraProperties = getDeploymentProperties();
+        String publisherURL = infraProperties.getProperty(PUBLISHER_URL);
+        String storeURL = infraProperties.getProperty(STORE_URL);
+        String warFileName = "jaxrs_basic";
+        String warFileLocation = System.getProperty("user.dir") + File.separator+ "src/test/resources" + File.separator
+                + warFileName + ".war";
+        String serviceEndpoint = "https://localhost:9443/services/";
+
+        if (publisherURL == null) {
```

Comment:
we do not need to do these in the test class now. Parent class does this. shall we remove?

Reply:
Removed